jni/src/faiss_index_service.cpp

```diff
@@ -86,15 +82,15 @@ jlong IndexService::initIndex(
     std::unique_ptr<faiss::Index> index(faissMethods->indexFactory(dim, indexDescription.c_str(), metric));
 
     // Set thread count if it is passed in as a parameter. Setting this variable will only impact the current thread
-    if(threadCount != 0) {
+    if (threadCount != 0) {
         omp_set_num_threads(threadCount);
     }
 
     // Add extra parameters that cant be configured with the index factory
     SetExtraParameters<faiss::Index, faiss::IndexIVF, faiss::IndexHNSW>(jniUtil, env, parameters, index.get());
 
     // Check that the index does not need to be trained
-    if(!index->is_trained) {
+    if (!index->is_trained) [[unlikely]] {
```

Comment:
Will we see benefit of this? Also, does this require c++ >= 20?

Reply:
It's C++11
Yeah, it's basically giving compiler hints so that it can reorder if branch.
I would say we won't lose anything by having it!

Reply:
Is there doc on this? Im seeing 20 here: https://en.cppreference.com/w/cpp/language/attributes/likely

Reply:
oh right. it's C++20. sorry for the confusion. :sweat_smile: 
will throw them away